chat-plugins/calculator.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+function isNumeric(str) {
+	return !isNaN(parseFloat(str)) && isFinite(str);
+}
+
+function clean(arr) {
+	let i;
+	for (i = 0; i < arr.length; i++) {
+		if (arr[i] === "") {
+			arr.splice(i, 1);
+		}
+	}
+	return arr;
+}
+
+function parseMathematicalExpression(infix) {
+	// Shunnting-yard Algorithm -- https://en.wikipedia.org/wiki/Shunting-yard_algorithm
+	let outputQueue = "";
+	let operatorStack = [];
+	let operators = {
+		"^": {
+			precedence: 4,
+			associativity: "Right",
+		},
+		"/": {
+			precedence: 3,
+			associativity: "Left",
+		},
+		"*": {
+			precedence: 3,
+			associativity: "Left",
+		},
+		"+": {
+			precedence: 2,
+			associativity: "Left",
+		},
+		"-": {
+			precedence: 2,
+			associativity: "Left",
+		},
+	};
+	infix = infix.replace(/\s+/g, "");
+	infix = infix.split(/([+\-*/^()])/);
+	infix = clean(infix);
+	let i;
+	for (i = 0; i < infix.length; i++) {
+		let token = infix[i];
+		if (isNumeric(token) === true) {
```

Comment:
You can just put this your last `else` on your `if` chain, instead of needing to run your `isNumeric` function.

Reply:
So something like this?: 
```	let i;
	for (i = 0; i < infix.length; i++) {
		let token = infix[i];
		if ("^*/+-".indexOf(token) !== -1) {
			let o1 = token;
			let o2 = operatorStack[operatorStack.length - 1];
			while ("^*/+-".indexOf(o2) !== -1 && ((operators[o1].associativity === "Left" && operators[o1].precedence <= operators[o2].precedence) || (operators[o1].associativity === "Right" && operators[o1].precedence < operators[o2].precedence))) {
				outputQueue.push(operatorStack.pop());
				o2 = operatorStack[operatorStack.length - 1];
			}
			operatorStack.push(o1);
		} else if (token === "(") {
			operatorStack.push(token);
		} else if (token === ")") {
			while (operatorStack[operatorStack.length - 1] !== "(") {
				outputQueue.push(operatorStack.pop());
			}
			operatorStack.pop();
		} else {
			outputQueue.push(token);
		}
	}

Reply:
I'm not sure I understand what you mean

Reply:
Yes, like that.